oak/server/rust/oak_runtime/src/channel.rs

```diff
@@ -222,17 +222,18 @@ impl ChannelReader {
         }
     }
 
-    /// Thread safe. Returns `Ok(true)` if there is at least one message in the channel. Fails with
-    /// `OakStatus::ERR_CHANNEL_CLOSED` if the underlying channel has been orphaned _and_ is empty.
-    // TODO(#566)
-    pub fn has_message(&self) -> Result<bool, OakStatus> {
+    /// Thread safe. This function will return:
+    /// - `READ_READY` if there is at least one message in the channel.
+    /// - `ORPHANED` if there are no messages and there are no writers
+    /// - `NOT_READ` if there are no messages but there are some writers
+    pub fn has_message(&self) -> ChannelReadStatus {
```

Comment:
Should this be renamed to `get_status` or something like that?

Reply:
Changed to `status`